### PR TITLE
post_max_size reduced from 16G to 10M

### PR DESCRIPTION
**Pull request by EParzefall**
Reducing post_max_size to a more reasonable value.
Backport to 10.12 and 10.13 necessary

### CODE REVIEW

File: modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc [hunk 1]
```diff
@@ -64,7 +64,7 @@ Set the following parameters inside the corresponding file using your own desire
 {userini}::
 [source,php]
 ----
-post_max_size=16G
+post_max_size=10G
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc:67]:
```suggestion
post_max_size=10M
```

File: modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc [hunk 2]
```diff
@@ -73,7 +73,7 @@ upload_tmp_dir=/mnt/php_big_temp/
 {htaccess}::
 [source,php]
 ----
-php_value	post_max_size=16G
+php_value	post_max_size=10G
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc:76]:
```suggestion
php_value	post_max_size=10M
```

File: modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc [hunk 3]
```diff
@@ -102,7 +102,7 @@ Set the following parameters inside the corresponding php.ini file using your ow
 
 [source,php]
 ----
-post_max_size=16G
+post_max_size=10G
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc:105]:
```suggestion
post_max_size=10M
```

File: modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc [hunk 4]
```diff
@@ -127,7 +127,7 @@ You can configure php parameters also per virtual host - if you have access to t
 	DocumentRoot /var/www/owncloud
 	ServerName myowncloud.com
 
-	php_admin_value	post_max_size 16G
+	php_admin_value	post_max_size 10G
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/files/big_file_upload_configuration.adoc:130]:
```suggestion
	php_admin_value	post_max_size 10M
```

**Comment by DeepDiver1975** [comment]
refs https://github.com/owncloud/security-tracker/issues/410

@EParzefall let me know if you have access to this .... most probably not .....

**Comment by DeepDiver1975** [comment]
out latest approach is to stick with the php production defaults:
 - set `upload_max_filesize` to 2MB https://github.com/php/php-src/blob/master/php.ini-production#L865
 - set `post_max_size`  to 8MB https://github.com/php/php-src/blob/master/php.ini-production#L713

We are verifying this by QA next week -  afterwards next steps are following ....

**Comment by EParzefall** [comment]
> refs https://github.com/owncloud/security-tracker/issues/410
> 
> @EParzefall let me know if you have access to this .... most probably not .....

@DeepDiver1975  Nope, no access.

**Comment by TheRealBro** [comment]
> out latest approach is to stick with the php production defaults:
> 
>     * set `upload_max_filesize` to 2MB https://github.com/php/php-src/blob/master/php.ini-production#L865
> 
>     * set `post_max_size`  to 8MB https://github.com/php/php-src/blob/master/php.ini-production#L713
> 
> 
> We are verifying this by QA next week - afterwards next steps are following ....

For the docker image we use 8MB for both values, just for the record, because otherwise it would be a BREAKING change with changing ENV variables.

**Comment by EParzefall** [comment]
Okay, I'll wait until you've reached a final decision after QA.

**Comment by iasdeoupxe** [comment]
What about all the core things which e.g. seems to use `513M` for both:

https://github.com/search?q=repo%3Aowncloud%2Fcore%20post_max_size&type=code

**Comment by DeepDiver1975** [comment]
We are still waiting for the verification by QA or whom ever ....

**Comment by mmattel** [comment]
> We are still waiting for the verification by QA or whom ever ....

Adding @jnweiger @GeraldLeikam

**Comment by jnweiger** [comment]
I don't have a good idea how to conduct manual tests here.
Is that needed? As long as it is just a documented recommodation, not much will break. 
But as soon as we update our docker containers with the lower limit, then any POST that tries something larger should break. 
I'd rely more on CI than on manual QA to find those...

**Comment by DeepDiver1975** [comment]
> I'd rely more on CI than on manual QA to find those...

That would require integration tests on all apps - which we simply don't have. Let me think about options ....

**Comment by DeepDiver1975** [comment]
> I don't have a good idea how to conduct manual tests here.
> Is that needed? As long as it is just a documented recommodation, not much will break.
> But as soon as we update our docker containers with the lower limit, then any POST that tries something larger should break.

doc change in this respect is as critical as docker image change - if too low functionalities will :boom:

**Comment by TheRealBro** [comment]
> > I don't have a good idea how to conduct manual tests here.
> > Is that needed? As long as it is just a documented recommodation, not much will break.
> > But as soon as we update our docker containers with the lower limit, then any POST that tries something larger should break.
> 
> doc change in this respect is as critical as docker image change - if too low functionalities will 💥

Doc change is maybe even more critical than changing docker image, which we did, as Collabora runs in a separate container and is not affected by changes. Changing settings on the host, if running Collabora on the same host, might blow up things. But still, we should move on with testing and/or decision.

**Comment by DeepDiver1975** [comment]
Just to be sure I explained it in detail enough - collabora and o365 (both using WOPI protocol on different api endpoints) are using POST to save the document back into owncloud.
As a result owncloud needs to accept POST request with a body of a few hundred MB.

Unfortunately there is no way to allow different post sizes on different endpoints .... at least I am not aware of any ....

**Pull request closed without merging**